layer/layer.go

```diff
@@ -188,6 +188,8 @@ type Store interface {
 	GetRWLayer(id string) (RWLayer, error)
 	GetMountID(id string) (string, error)
 	ReleaseRWLayer(RWLayer) ([]Metadata, error)
+	LockRWLayer()
+	UnlockRWLayer()
```

Comment:
@yummypeng Wouldn't you need to lock during for each invocation of `ls.driver.Remove`?